nbs/05_rules.ipynb

```diff
@@ -10,7 +10,7 @@
   },
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #135.            # Nodes in P do NOT add attributes that aren't in the corresponding LHS node(s).
</pre><p>do we really need to do these checks in a separate block? cant we make a set of attribute per logical variable for both regular nodes and collections and do this check once for both?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/DeanLight/graph_rewrite/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Reply:
<p>There is the following line:</p><p>'''lhs_attrs = set(self.lhs.nodes(data=True)[node_lhs].keys())</p><p> </p><p>In the first check we need to use self.lhs and in the second one we need to use self.collections</p><p> </p><p>I suppose we can create a dict from each node to its attributes that include both sets and make it more elegant</p>

Reply:
<p>for Stav - here is an example of how this should be done:</p><p> </p><p>node_attr_dict = {</p><p>		&nbsp;&nbsp;&nbsp;&nbsp;node:set(self.lhs.nodes(data=True)[node_lhs].keys()) for ...</p><p>	} | {</p><p>		&nbsp;&nbsp;&nbsp;&nbsp;node:set(self.collections.nodes(data=True)[node_lhs].keys())</p><p>		&nbsp;&nbsp;&nbsp;&nbsp;for ....</p><p>	}</p>